app/frontend/wallet/cardano-wallet-secret-crypto-provider.ts

```diff
@@ -52,6 +52,7 @@ const CardanoWalletSecretCryptoProvider = (
     const result = derivePrivate(hdNode.toBuffer(), childIndex, state.derivationScheme.number)
 
     return HdNode({
+      secret: null,
```

Comment:
Instead of this hack, we should have `HdNodeFromSecret` or something like that. Also, `secret` is probably wrong parameter name, if I recall the code, this should be more like `compact` (maybe `HdNodeFromCompactedForm` or something like that?)